ipam/ring/ring.go

```diff
@@ -524,7 +562,20 @@ func (r *Ring) PickPeerForTransfer(isValid func(mesh.PeerName) bool) mesh.PeerNa
 // Transfer will mark all entries associated with 'from' peer as owned by 'to' peer
 // and return ranges indicating the new space we picked up
 func (r *Ring) Transfer(from, to mesh.PeerName) []address.Range {
+	return r.transfer(from, to, false)
+}
+
+// Does Transfer from a dead peer
+func (r *Ring) AdminTransfer(from, to mesh.PeerName) []address.Range {
+	return r.transfer(from, to, true)
+}
+
+func (r *Ring) transfer(from, to mesh.PeerName, isAdmin bool) []address.Range {
 	r.assertInvariants()
+	defer r.trackUpdates()()
+	if isAdmin {
+		defer r.trackUpdatesOfPeer(from)()
```

Comment:
Just to recap, this would make `Transfer` to call both:

```go
defer r.trackUpdates()()
defer r.trackUpdatesOfPeer(from)()
```

In the case of `alloc.Shutdown()`, the second defer statement (which is executed before the first defer statement) will do the necessary cleanup (i.e. will remove the rules). Whilst, the first defer statement will call `tracker.HandleUpdate(nil, nil)` resulting in no action.

In the case of `alloc.AdminTakeoverRanges(...)`, the second defer statement will remove non-local rules belonging to the `from` peer and afterwards, the first defer statement will add new rules for the local peer.